src/services/session.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const protocolCallbacks = require('@services/protocolCallbacks/')
+const sessionAttendanceQueries = require('@database/storage/sessionAttendance/queries')
+const userQueries = require('@database/storage/user/queries')
+const bapQueries = require('@database/storage/bap/queries')
+const crypto = require('crypto')
+
+exports.session = async (requestBody) => {
+	try {
+		const sessionId = requestBody.sessionId
+		const sessionAttendance = await sessionAttendanceQueries.findBySessionId(sessionId)
+		if (!sessionAttendance) {
+			return console.log('SessionAttendance Not Found')
+		}
+		const userIds = sessionAttendance.map((sessionAttendee) => sessionAttendee.userId)
+		const users = await userQueries.findByIds(userIds)
+		if (!users) {
+			return console.log('Users Not Found')
+		}
+		const bapIds = users.map((user) => user.bapId)
+		const baps = await bapQueries.findByIds(bapIds)
+		if (!baps) {
+			return console.log('BAP Not Found')
+		}
+		let usersWithBapAndAttendance = users.map((user) => {
+			const bapInfo = baps.find((f) => f._id.toString() === user.bapId.toString())
+			const sessionAttendanceInfo = sessionAttendance.find((f) => f.userId.toString() === user._id.toString())
+			return {
+				...user,
+				bapInfo,
+				sessionAttendanceInfo,
+			}
+		})
+		usersWithBapAndAttendance.map(async (user) => {
```

Comment:
When you are switching back to regular responses, a question arises about when to give back a success response to mentoring. Do we do that when all the requests being generated was successful or can we give a success response as soon as we trigger these requests (but without considering if any of them were in-fact successful). Please bring this up as a discussion point with the team. 

From my perspective, we shouldn't be checking each and every request to be successful (Feels fragile). If team agrees, you can use promise.race to split the execution and give back the control to controller while requests are still being generated.